### PR TITLE
Added test for fix in Split layer

### PR DESCRIPTION
**Pull request by czgdp1807**
See, https://github.com/opencv/opencv/pull/16424#issuecomment-578165672

ping @dkurt Please take a look. Please let me know if there is any change to make.

### CODE REVIEW

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 1]
```diff
@@ -426,6 +426,26 @@ def forward(self, x):
...
+
+model = Split()
+input = Variable(torch.randn([1., 2., 3., 4., 5., 6.]))
+save_data_and_model("split_1", input, model)
+
+model = Split(0, [2, 4])
+save_data_and_model("split_2", input, model)
+
+model = Split(0)
+save_data_and_model("split_3", input, model)
```
Comment (dkurt) [inline testdata/dnn/onnx/generate_onnx_models.py]: What's the difference between `split_1` and `split_3` cases?

Comment (czgdp1807) [reply testdata/dnn/onnx/generate_onnx_models.py]: I was trying to tests how the model will be processed by ONNXImporter in case no attribute is given to `Split` in `split_1`. In `split_3`, I am trying to test the case when only axis is given. I think one more case is left, when only split is given. I will add that too.

**Comment by czgdp1807** [comment]
Update:
Hi,
I will be having my mid semester examinations till 7th February(IST). So the work on this may get slowed down. Please do not close this PR until then. I am on it and I will put efforts to complete it from 7th Feb. Thank you and apologies.

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 2]
```diff
@@ -426,6 +426,30 @@ def forward(self, x):
...
+save_data_and_model("split_1", input, model)
+
+model = Split(dim=0)
+save_data_and_model("split_2", input, model)
+
+model = Split(split_size_sections=[2, 4])
+save_data_and_model("split_3", input, model)
+
+model = Split(dim=0, split_size_sections=[2, 4])
+save_data_and_model("split_4", input, model)
```
Comment (dkurt) [inline testdata/dnn/onnx/generate_onnx_models.py]: Looks like files are missed.

Comment (czgdp1807) [reply testdata/dnn/onnx/generate_onnx_models.py]: Yes, in the latest logs, `findData` is unable to find the files.

Comment (dkurt) [reply testdata/dnn/onnx/generate_onnx_models.py]: So this PR should include these files

Comment (czgdp1807) [reply testdata/dnn/onnx/generate_onnx_models.py]: Actually, `onnx.checker.check_model(model_def)` is throwing `Segmentation Fault` on my system. Probably that's why code never reaches to the above lines and the already present files are detected but new ones aren't generated. I will add my `onnx` files separately and will push here.

Comment (czgdp1807) [reply testdata/dnn/onnx/generate_onnx_models.py]: > So this PR should include these files

Done. Please take a look.

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 3]
```diff
@@ -426,6 +426,31 @@ def forward(self, x):
...
+
+    def __init__(self, *args, **kwargs):
+        super(Split, self).__init__()
+        self.split_size_sections = \
+            kwargs.get('split_size_sections', 1)
+        self.dim = kwargs.get('dim', 0)
+
+    def forward(self, x):
+        tup = torch.split(x, self.split_size_sections, self.dim)
+        return torch.cat(tup)
```
Comment (czgdp1807) [inline testdata/dnn/onnx/generate_onnx_models.py]: I have used this because, usually, `torch.split` returns a python `tuple` which if converted to a tensor here will render the code useless as the layer will be saved as `Constant` and not `Split`.
And we cannot save python tuple in `npy` file as they are not produced correctly in the `cpp` tests and moreover they will require a lot of changes across some files to make tuples work in the code. Hence, this trick tests the `Split` layer with minimal changes in the code. Though I am open to suggestions on improving this.

Comment (czgdp1807) [reply testdata/dnn/onnx/generate_onnx_models.py]: I have run the tests locally they, pass on my system.

**Comment by czgdp1807** [comment]
The tests passed. Is it good to go?
